app/scripts/metamask-controller.js

```diff
@@ -4062,6 +4055,17 @@ export default class MetamaskController extends EventEmitter {
     this.sendUpdate();
   }
 
+  //=============================================================================
+  // REQUEST QUEUE
+  //=============================================================================
+
+  setUseRequestQueue(value) {
+    this.preferencesController.setUseRequestQueue(value);
+    this.selectedNetworkController.update((state) => {
+      state.perDomainNetwork = value;
+    });
```

Comment:
i think this has to be called with the current useRequestQueue value right after selectedNetworkController init

Reply:
actually nevermind